src/dataprod_config.jl

```diff
@@ -279,83 +292,127 @@ function analysis_runs end
+            @NamedTuple{period::DataPeriod, run::DataRun, cal::nttype, phy::nttype, fft::nttype, cos::nttype, lac::nttype}((period, run, get_cat_entry(:cal), get_cat_entry(:phy), get_cat_entry(:fft), get_cat_entry(:cos), get_cat_entry(:lac)))
         end
-        periods_and_runs = [[make_row(p, r, ri) for (r, ri) in rs] for (p, rs) in rinfo]
-        flat_pr = sort(StructArray(vcat(periods_and_runs...)::Vector{@NamedTuple{period::DataPeriod, run::DataRun, cal::nttype, phy::nttype, fft::nttype}}))
-        Table(merge(columns(flat_pr), (cal = Table(StructArray(flat_pr.cal)), phy = Table(StructArray(flat_pr.phy)), fft = Table(StructArray(flat_pr.fft)))))
+        periods_and_runs = [[make_row(p, r, ri) for (r, ri) in rs if any((cat -> (cat, p, DataRun(r)) in allowed), (:cal, :phy, :fft, :cos, :lac))] for (p, rs) in rinfo]
+        flat_pr = sort(StructArray(vcat(filter(!isempty, periods_and_runs)...)::Vector{@NamedTuple{period::DataPeriod, run::DataRun, cal::nttype, phy::nttype, fft::nttype, cos::nttype, lac::nttype}}))
+        Table(merge(columns(flat_pr), (cal = Table(StructArray(flat_pr.cal)), phy = Table(StructArray(flat_pr.phy)), fft = Table(StructArray(flat_pr.fft)), cos = Table(StructArray(flat_pr.cos)), lac = Table(StructArray(flat_pr.lac)))))
     end
 end
 export runinfo
 
-runinfo(data::LegendData, period::DataPeriodLike) = runinfo(data) |> filterby(@pf $period == DataPeriod(period))
+runinfo(data::LegendData, period::DataPeriodLike; runlist::Symbol = :valid) = runinfo(data; runlist) |> filterby(@pf $period == DataPeriod(period))
 
-function runinfo(data::LegendData, runsel::RunSelLike)
+function runinfo(data::LegendData, runsel::RunSelLike; runlist::Symbol = :valid)
     period, run = runsel
     period, run = DataPeriod(period), DataRun(run)
-    t = runinfo(data) |> filterby(@pf $period == period && $run == run)
+    t = runinfo(data; runlist) |> filterby(@pf $period == period && $run == run)
     if isempty(t)
         throw(ArgumentError("No run information found for period $period run $run"))
     else
         Table(t)
     end
 end
 
-function runinfo(data::LegendData, runsel::RunCategorySelLike)
+function runinfo(data::LegendData, runsel::RunCategorySelLike; runlist::Symbol = :valid)
     period, run, category = runsel
     period, run, category = DataPeriod(period), DataRun(run), DataCategory(category)
-    getproperty(runinfo(data, (period, run)), Symbol(category))
+    getproperty(runinfo(data, (period, run); runlist), Symbol(category))
 end
-runinfo(data, fk::FileKey) = runinfo(data, (fk.period, fk.run, fk.category))
-runinfo(data, selectors...) = runinfo(data, selectors)
+runinfo(data, fk::FileKey; runlist::Symbol = :valid) = runinfo(data, (fk.period, fk.run, fk.category); runlist)
+runinfo(data, selectors...; runlist::Symbol = :valid) = runinfo(data, selectors; runlist)
```

Comment:
Instead of defining the same keyword argument default, you could just pass the keywords arguments through. 
Then, the base `runinfo` will either set the default or take whatever it was given.
This should then be explained in the `runinfo` docstring.

```suggestion
runinfo(data, fk::FileKey; kwargs...) = runinfo(data, (fk.period, fk.run, fk.category); kwargs...)
runinfo(data, selectors...; kwargs...) = runinfo(data, selectors; kwargs...)
```